dags/operators/task_sensor.py

```diff
@@ -0,0 +1,108 @@
+            'Poking for %s.%s on %s ... ',
+            self.external_dag_id, self.external_task_id, serialized_dttm_filter
+        )
+
+        DM = DagModel
+        TI = TaskInstance
+        DR = DagRun
+        if self.check_existence:
+            dag_to_wait = session.query(DM).filter(
+                DM.dag_id == self.external_dag_id
+            ).first()
+
+            if not dag_to_wait:
+                raise AirflowException('The external DAG '
+                                       '{} does not exist.'.format(self.external_dag_id))
+            else:
+                if not os.path.exists(dag_to_wait.fileloc):
+                    raise AirflowException('The external DAG '
+                                           '{} was deleted.'.format(self.external_dag_id))
+
+            if self.external_task_id:
+                refreshed_dag_info = DagBag(dag_to_wait.fileloc).get_dag(self.external_dag_id)
+                if not refreshed_dag_info.has_task(self.external_task_id):
+                    raise AirflowException('The external task'
+                                           '{} in DAG {} does not exist.'.format(self.external_task_id,
+                                                                                 self.external_dag_id))
+        
+        # custom implementation to check for failed tasks
+        if self.external_task_id:
+            # .count() is inefficient
+            count_allowed = session.query(func.count()).filter(
+                TI.dag_id == self.external_dag_id,
+                TI.task_id == self.external_task_id,
+                TI.state.in_(self.allowed_states),
+                TI.execution_date.in_(dttm_filter),
+            ).scalar()
+
+            count_failed = session.query(func.count()).filter(
+                TI.dag_id == self.external_dag_id,
+                TI.task_id == self.external_task_id,
+                TI.state.in_(self.failed_states),
+                TI.execution_date.in_(dttm_filter),
+            ).scalar()
+        else:
+            # .count() is inefficient
+            count_allowed = session.query(func.count()).filter(
+                DR.dag_id == self.external_dag_id,
+                DR.state.in_(self.allowed_states),
+                DR.execution_date.in_(dttm_filter),
+            ).scalar()
+
+            count_failed = session.query(func.count()).filter(
+                TI.dag_id == self.external_dag_id,
+                TI.task_id == self.external_task_id,
+                TI.state.in_(self.failed_states),
+                TI.execution_date.in_(dttm_filter),
+            ).scalar()
+
+        if count_failed == len(dttm_filter):
+            if self.external_task_id:
+                raise AirflowException(
+                    f'The external task {self.external_task_id} in DAG {self.external_dag_id} failed.'
+                )
+            else:
+                raise AirflowException(f'The external DAG {self.external_dag_id} failed.')
```

Comment:
This should be functionally equivalent to the Airflow 2 [`ExternalTaskSensor` implementation](https://github.com/apache/airflow/blob/8505d2f0a4524313e3eff7a4f16b9a9439c7a79f/airflow/sensors/external_task.py#L165)